sources/spanner/infoschema.go

```diff
@@ -349,19 +350,31 @@ func (isi InfoSchemaImpl) GetIndexes(conv *internal.Conv, table common.SchemaAnd
 		if err != nil {
 			return nil, fmt.Errorf("couldn't read row while fetching interleaved tables: %w", err)
 		}
-		err = row.Columns(&name, &column, &sequence, &ordering, &isUnique)
-		if err != nil {
-			fmt.Println(err)
-			conv.Unexpected(fmt.Sprintf("Can't scan: %v", err))
-			continue
+		if isi.SpDialect == constants.DIALECT_POSTGRESQL {
+			err = row.Columns(&name, &column, &sequence, &ordering, &isPgUnique)
+			if err != nil {
+				fmt.Println(err)
+				conv.Unexpected(fmt.Sprintf("Can't scan: %v", err))
```

Comment:
If an `err` is encountered, we still go ahead with constructing the `conv` object.

Can this lead to issues? For example, if an error is encountered while reading indexes from the source database, that index will never get migrated during the schema conversion.

I wonder what should be our strategy to handle these failures...should we -

1. Completely fail and prevent user from moving forward.
2. Should we do a best effort conversion and inform the user that there were partial errors in reading the schema.
3. If there are partial errors, should we store this information in `conv` or someplace else - this can be used has a hook later to a subsequent action (example - refuse validations if the spanner schema was created using a partial error conv?)

We need to think more on this front. Please create a tech debt item with all the relevant details.

Reply:
created a ticket with id : 315784390